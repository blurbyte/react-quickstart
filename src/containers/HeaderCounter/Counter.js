//presentational component
import React from 'react';
import PropTypes from 'prop-types';

//components
import NavLink from '../../components/NavLink';
import Wrapper from './Wrapper';
import CounterValue from './CounterValue';

const Counter = ({ counter }) => (
  <Wrapper>
    <NavLink to="/counter">Manage example counter</NavLink>
    <CounterValue>{counter}</CounterValue>
  </Wrapper>
);

Counter.propTypes = {
  counter: PropTypes.number.isRequired
};

export default Counter;
