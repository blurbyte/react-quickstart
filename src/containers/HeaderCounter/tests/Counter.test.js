import React from 'react';
import renderer from 'react-test-renderer';

import Counter from '../Counter';

// mocked components
jest.mock('components/NavLink', () => 'NavLink');
jest.mock('../Wrapper', () => 'Wrapper');
jest.mock('../CounterValue', () => 'CounterValue');

describe('<Counter />', () => {
  it('should render correctly', () => {
    // props init
    const props = {
      counter: 10
    };

    // component init
    const component = renderer.create(<Counter {...props} />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
