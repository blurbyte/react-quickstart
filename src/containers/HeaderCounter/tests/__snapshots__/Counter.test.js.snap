// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> should render correctly 1`] = `
<Wrapper>
  <NavLink
    to="/counter"
  >
    Manage example counter
  </NavLink>
  <CounterValue>
    10
  </CounterValue>
</Wrapper>
`;
